twine/package.py

```diff
@@ -75,7 +75,7 @@ def __init__(
         self.metadata = metadata
         self.python_version = python_version
         self.filetype = filetype
-        self.safe_name = _safe_name(metadata.name)
+        self.safe_name = _safe_name(metadata.name or "")
```

Comment:
Not sure why version and name can be None now in typing, I thought those were always set?